Library/Formula/go.rb

```diff
@@ -16,72 +16,39 @@ class Go < Formula
   option "without-cgo", "Build without cgo"
   option "without-godoc", "godoc will not be installed for you"
   option "without-vet", "vet will not be installed for you"
 
-  deprecated_option "cross-compile-all" => "with-cc-all"
-  deprecated_option "cross-compile-common" => "with-cc-common"
 
   resource "gotools" do
     url "https://go.googlesource.com/tools.git",
-    :revision => "69db398fe0e69396984e3967724820c1f631e971"
+    :revision => "d02228d1857b9f49cd0252788516ff5584266eb6"
   end
 
   resource "gobootstrap" do
-    if MacOS.version > :lion
-      url "https://storage.googleapis.com/golang/go1.4.2.darwin-amd64-osx10.8.tar.gz"
-      sha256 "c2f53983fc8fe5159d811081022ebc401b8111759ce008f91193abdae82cdbc9"
-    else
-      url "https://storage.googleapis.com/golang/go1.4.2.darwin-amd64-osx10.6.tar.gz"
-      sha256 "da40e85a2c9bda9d2c29755c8b57b8d5932440ba466ca366c2a667697a62da4c"
-    end
+    url "https://storage.googleapis.com/golang/go1.5.darwin-amd64.tar.gz"
+    sha256 "fff51671abfbe3e68109d8bc5c7ee66d2da17fb1235ed834ce7fd294d607803f"
   end
 
   def install
-    # host platform (darwin) must come last in the targets list
-    if build.with? "cc-all"
-      targets = [
-        ["linux",   ["386", "amd64", "arm"]],
-        ["freebsd", ["386", "amd64", "arm"]],
-        ["netbsd",  ["386", "amd64", "arm"]],
-        ["openbsd", ["386", "amd64"]],
-        ["windows", ["386", "amd64"]],
-        ["dragonfly", ["386", "amd64"]],
-        ["plan9",   ["386", "amd64"]],
-        ["solaris", ["amd64"]],
-        ["darwin",  ["386", "amd64"]]
-      ]
-    elsif build.with? "cc-common"
-      targets = [
-        ["linux",   ["386", "amd64", "arm"]],
-        ["windows", ["386", "amd64"]],
-        ["darwin",  ["386", "amd64"]]
-      ]
-    else
-      targets = [["darwin", [""]]]
-    end
 
-    if build.head? || build.devel?
-      # GOROOT_FINAL must be overidden later on real Go install
-      ENV["GOROOT_FINAL"] = buildpath/"gobootstrap"
+    # GOROOT_FINAL must be overidden later on real Go install
+    ENV["GOROOT_FINAL"] = buildpath/"gobootstrap"
 
-      # build the gobootstrap toolchain Go >=1.4
-      (buildpath/"gobootstrap").install resource("gobootstrap")
-      cd "#{buildpath}/gobootstrap/src" do
-        system "./make.bash", "--no-clean"
-      end
-      # This should happen after we build the test Go, just in case
-      # the bootstrap toolchain is aware of this variable too.
-      ENV["GOROOT_BOOTSTRAP"] = ENV["GOROOT_FINAL"]
+    # build the gobootstrap toolchain Go >=1.4
+    (buildpath/"gobootstrap").install resource("gobootstrap")
+    cd "#{buildpath}/gobootstrap/src" do
+      system "./make.bash", "--no-clean"
     end
+    # This should happen after we build the test Go, just in case
+    # the bootstrap toolchain is aware of this variable too.
+    ENV["GOROOT_BOOTSTRAP"] = ENV["GOROOT_FINAL"]
 
     cd "src" do
       targets.each do |os, archs|
```

Comment:
You've removed all the logic for `targets` so `targets.each do |os, archs|` is pointing at nothing. You'll need to tweak this line for whatever is necessary now.